sky/backends/cloud_vm_ray_backend.py

```diff
@@ -699,6 +699,11 @@ def _yield_region_zones(self, to_provision: resources_lib.Resources,
                             f'{handle.cluster_yaml} '
                             'has been changed from '
                             f'{prev_resources.region} to {region}.')
+                    if zones != prev_resources.zones:
+                        raise ValueError(f'Zones mismatch. The zones in '
+                                         f'{handle.cluster_yaml} '
+                                         'have been changed from '
+                                         f'{prev_resources.zones} to {zones}.')
```

Comment:
Thanks for catching this. It's fixed now. But I wonder the reason behind checking between Ray YAML zone field and the zone in the handle. Is this because we want to avoid user accidentally modifying the Ray YAML file?